### ISSUE TITLE
Timeout handling is unclear: connection timeout or plugin runtime/execution timeout?

### ISSUE THREAD

**Issue by atc0005** [atc0005/check-vmware#414]
TL;DR: Code indicates that the `timeout` setting is for *connections*, the documentation seems to indicate that it is for plugin *runtime* or *execution*. One of these needs to be fixed and it looks like the code should be changed.

---

I recently tested plugins from this repo against a VMware cluster with 200+ VMs. While working with the `check_vmware_tools` plugin I ran into `Context Deadline Exceeded` errors. I worked around it by increasing the value to around 30 seconds. This wasn't viable long-term however as the default Nagios plugin timeout is 30 seconds (if I recall correctly).

While GH-399/GH-404 resolved the root problem for that scenario, the problem remains:

- documentation currently indicates that the `-t` and `--timeout` flags are used for plugin execution
- the `internal/config` package indicates that the timeout value is used as the *connection* timeout

Snippets:

https://github.com/atc0005/check-vmware/blob/94b4386c05520bd779711e069801cd6ae5d6bb57/internal/config/getters.go#L15-L20

https://github.com/atc0005/check-vmware/blob/94b4386c05520bd779711e069801cd6ae5d6bb57/internal/config/flags.go#L280-L281

https://github.com/atc0005/check-vmware/blob/94b4386c05520bd779711e069801cd6ae5d6bb57/cmd/check_vmware_tools/main.go#L74-L75

This last snippet shows the creation of the base `ctx` value which is passed into functions/methods which in turn provide it as required to `vmware/govmomi` functions.

I'm currently thinking that I should review the documentation to make sure that it consistently indicates that the `timeout` setting is used for plugin execution, then update the codebase to consistently note that it's not a connection timeout, but instead a plugin runtime/execution timeout.

If we need a separate timeout specific to the initial connection attempt we can add a new flag to cover that case.
